### PR TITLE
[image_picker_android] - will fix crash on Android 12+ devices

### PR DESCRIPTION
**Pull request by Mairramer**
This will fix https://github.com/flutter/flutter/issues/147280

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/image_picker/image_picker_android/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.8.12
+
+* Fixes app crashes on Android 12+ caused by selected images with size 0.
```
Comment (gmackall) [inline packages/image_picker/image_picker_android/CHANGELOG.md:3]: nit 
```suggestion
* Fixes app crashes on Android 12+ caused by selecting images with size 0.
```

**Review by gmackall — APPROVED** [review]
LGTM! Thanks for the fix

**Comment by gmackall** [comment]
Will need a test, though (you'd probably want to add a case in `ImagePickerDelegateTest` that invokes `onActivityResult` in a way that triggers this code path)

**Comment by Mairramer** [comment]
> Test LGTM, are you able to add the autosubmit label (I can't remember what permissions non-member contributors have)?
> 
> If yes feel free to land at your convenience, otherwise let me know and I can land it

Unfortunately I don't have permission to add autosubmit label.

**Review by tarrinneal — APPROVED** [review]

**Comment by stuartmorgan** [comment]
FYI this only fixed the get-media codepath. `handleChooseMultiImageResult` still has the same bug. We should really extract a helper for this as @gmackall suggested in the issue.

**Pull request merged**